src/main/antora/modules/ROOT/pages/repositories/projections.adoc

```diff
@@ -4,3 +4,28 @@
 include::{commons}@data-commons::page$repositories/projections.adoc[leveloffset=+1]
 
 NOTE: It is important to note that <<projections.dtos,Class-based projections>> with JPQL is limited to *constructor expressions* in your JPQL expression, e.g. `SELECT new com.example.NamesOnly(u.firstname, u.lastname) from User u`. (Note the usage of a FQDN for the DTO type!) This JPQL expression can be used in `@Query` annotations as well where you define any named queries. And it's important to point out that class-based projections do not work with native queries AT ALL. As a workaround you may use named queries with `ResultSetMapping` or the Hibernate specific https://docs.jboss.org/hibernate/orm/6.0/javadocs/org/hibernate/transform/ResultTransformer.html[`ResultTransformer`]
+
+[NOTE]
+====
+Some JPA providers may require additional hints when working with <<projections.dtos,Class-based projections>> especially when using those directly with method derived queries.
```

Comment:
That confuses me. `@PersistenceCreator` is a Spring Data annotation, I'm sure that JPA providers don't care about that. 

I would expect that the situation is more: We need either a constructor taking all the properties as arguments and if it isn't the only constructor we need it to be marked with `@PersistenceCreator`.

Reply:
It's our code that determines input properties. We use our mechanism to discover whether there is a preferred constructor and the properties it takes.